engine/schema/src/com/cloud/network/dao/LoadBalancerDaoImpl.java

```diff
@@ -75,4 +84,40 @@ protected LoadBalancerDaoImpl() {
         return listBy(sc);
     }
 
+    @Override
+    public boolean isLoadBalancerRulesMappedToVmGuestIp(long instanceId, String instanceIp, long networkId, FirewallRule.Purpose purpose)
+    {
+        SearchBuilder<LoadBalancerVMMapVO> lbVmMapSearch = _loadBalancerVMMapDao.createSearchBuilder();
+        lbVmMapSearch.and("instanceIp", lbVmMapSearch.entity().getInstanceIp(),SearchCriteria.Op.EQ);
+        lbVmMapSearch.and("instanceId", lbVmMapSearch.entity().getInstanceId(), SearchCriteria.Op.EQ);
+
+        SearchBuilder<FirewallRuleVO> firewallRuleIdSearch = _firewallRulesDao.createSearchBuilder();
```

Comment:
You might know LoadBalancerVO extends FirewallRuleVO, you can use LoadBalancerVO instead.